addon/components/popover-on-element.js

```diff
@@ -64,6 +65,42 @@ export default TooltipAndPopoverComponent.extend({
 
     }
   },
+  willDestroyElement() {
+    this._super(...arguments);
+
+    const event = this.get('event');
+    const target = this.get('target');
+    const $target = $(target);
+    const $popover = this.$();
+    const _showOn = this.get('_showOn');
+
+    if (event === 'none') {
+      return;
+    }
+
+    if (_showOn === 'mouseenter') {
```

Comment:
I think I'll be able to simplify the `_showOn == 'mouseenter'` case now that I only use _isMouseInside for that approach.